packages/common/src/types.ts

```diff
@@ -319,11 +348,25 @@ export interface IHubContent extends IHubResource, IItem {
   data?: {
     [propName: string]: any;
   };
+  // service types: Feature Service, Map Service
+  /** service information (currentVersion, capabilities, maxRecordCount etc) */
+  server?: Partial<IFeatureServiceDefinition>;
+  /** layer information (geometryType, fields, etc) for related layers in the service */
+  layers?: Array<Partial<ILayerDefinition>>;
+  // layer types: Feature Layers, Raster Layers
+  /** layer information (geometryType, fields, etc) */
+  layer?: Partial<ILayerDefinition>;
+  recordCount?: number;
```

Comment:
how does `recordCount` work w/ `layers`?

Reply:
`recordCount` only refers the the "current" layer (see https://github.com/Esri/hub.js/pull/545#discussion_r652014708). and is not returned for service types (`Feature Service`, `Map Service`), only for the `Feature Layer` pseudo type.

Reply:
i.e it must be re-fetched if you switch layers